src/transformers/trainer.py

```diff
@@ -3716,3 +3720,45 @@ def _gather_and_numpify(self, tensors, name):
             tensors = distributed_concat(tensors)
 
         return nested_numpify(tensors)
+
+    def _add_sm_patterns_to_gitignore(
+        self, patterns: List[str] = [
+            "*.sagemaker-uploading",
+            "*.sagemaker-uploaded"
+        ]) -> None:
+        """Add SageMaker Checkpointing patterns to .gitignore file.
+
+        Args:
+            patterns (List[str], optional): List of patterns to add to .gitignore. Defaults to
+                [".sagemaker-uploading", ".sagemaker-uploaded"].
+        """
+        # Get current .gitignore content
```

Comment:
Make sure we are on the main process here (we should be since we return early in `init_git_repo` but we could call this method from somewhere else).